### ISSUE TITLE
GPU temperature sensor (rx580 4gb) missing after update

### ISSUE THREAD

**Issue by BlackSharp** [Rem0o/FanControl.Releases#2509]
My SAPPHIRE NITRO+ Radeon RX 580 4GB Radeon RX 580 temperature gauge is missing after the latest update
It always worked before.
![FanControl_ax4dlIcVKa](https://github.com/Rem0o/FanControl.Releases/assets/16515416/77441c7a-d764-4528-a580-ae1e8f2f9887)
![o2kMp28cUL](https://github.com/Rem0o/FanControl.Releases/assets/16515416/f1d6fe12-1f36-4788-9e7c-2222f895f122)

[Log.txt](https://gist.github.com/BlackSharp/0072346e0f1a3adbc0763e564d1f9fad)

hardware: 
![HWiNFO64_rwbcNoyEwT](https://github.com/Rem0o/FanControl.Releases/assets/16515416/b59b0a19-4705-499a-a535-f263c3948199)

**Comment by Lana-chan** [comment]
same issue for me on RX480

**Comment by Rem0o** [comment]
Also, in the meantime, give https://github.com/Rem0o/FanControl.ADLX a go.   
Untick the GPU section in sensor settings, and keep ADLX.

**Comment by Lana-chan** [comment]
the missing items were `/gpu-amd/0/fan/0` and `/gpu-amd/0/control/0`, which were present in FanControl V189 but missing in V190 and the linked LHM, so probably removed in newer versions of LHM?

[LibreHardwareMonitor.Report.txt](https://github.com/Rem0o/FanControl.Releases/files/15312564/LibreHardwareMonitor.Report.txt)

ADLX plugin does give proper sensor and control for my GPU, so my issue is solved by this, thank you

**Comment by BlackSharp** [comment]
[LibreHardwareMonitor.Report.txt](https://github.com/Rem0o/FanControl.Releases/files/15312656/LibreHardwareMonitor.Report.txt)

**Comment by Happymax** [comment]
> Also, in the meantime, give https://github.com/Rem0o/FanControl.ADLX a go. Untick the GPU section in sensor settings, and keep ADLX.

For me ADLX also cannot detect any sensors (GPU source from LHM disabled). I'm using an RX480.

**Comment by epinter** [comment]
> the missing items were `/gpu-amd/0/fan/0` and `/gpu-amd/0/control/0`, which were present in FanControl V189 but missing in V190 and the linked LHM, so probably removed in newer versions of LHM?
> 
> [LibreHardwareMonitor.Report.txt](https://github.com/Rem0o/FanControl.Releases/files/15312564/LibreHardwareMonitor.Report.txt)
> 
> ADLX plugin does give proper sensor and control for my GPU, so my issue is solved by this, thank you

Thanks for the log. The sensors were not removed. There was an update to support the sensors for RX 7000 series. The sensors used by Polaris or the code to support them didn't change, I'm not sure yet why they disappeared.

**Comment by epinter** [comment]
Try this build of LHM, tell me if it works (if not, please generate another txt report). Download links at the end of the page, LibreHardwareMonitor-net472 is full LHM, others are the libraries for other .net versions.

https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/actions/runs/9085676231

**Comment by Lana-chan** [comment]
that fixed version does report temperature and fans on my machine

**Comment by WATEVRBRO** [comment]
The ADLX does not give proper sensor or control for my GPU (NITRO+ RX 580 8G G5 SE).

I am looking at the https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/actions/runs/9085676231 but i am not sure I understand how to install this. GitHub is new to me and the UI is difficult and the javascript instructions are confusing to me.

here is my txt report
[LibreHardwareMonitor.Report.txt](https://github.com/Rem0o/FanControl.Releases/files/15360684/LibreHardwareMonitor.Report.txt)

**Comment by Rem0o** [comment]
@WATEVRBRO  I updated FanControl to V191 to include the new LHM release.

**Comment by weinitom** [comment]
> @WATEVRBRO I updated FanControl to V191 to include the new LHM release.

Hi, thanks for the fast update. I thought it might fix my problem also, but even with V191 (and V190) the temperature of my AMD R9 390 is still missing. Also I had no luck with ADLX. Could there be anything else wrong?

**Comment by weinitom** [comment]
> Try this build of LHM, tell me if it works (if not, please generate another txt report). Download links at the end of the page, LibreHardwareMonitor-net472 is full LHM, others are the libraries for other .net versions.
> 
> https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/actions/runs/9085676231

Hi, thanks for the quick update. I thought it might also solve my problem, but even with V191 (and V190) the temperature of my AMD R9 390 is still not available. But I got the fan control back with V191. I also had no luck with ADLX. Could there be something else wrong? Worked perfectly with the versions before V190. Thanks for the great work btw, great tool!

**Comment by Rem0o** [comment]
@weinitom @epinter 
Might be, I believe collectively we tested with RX 400/500, RX 5000/6000/7000, but we didn't go older.   The required gymnastic to support every generation might not be perfect yet.    My guess @epinter is that using the ADL2 calls works for RX 400/500, but not for older cards, which still requires the old ADL calls.  You'd have to kinda cascade every possible version of the ADL functions from newest to oldest until one registers something to support every possible generation.

**Comment by epinter** [comment]
> @weinitom @epinter Might be, I believe collectively we tested with RX 400/500, RX 5000/6000/7000, but we didn't go older. The required gymnastic to support every generation might not be perfect yet. My guess @epinter is that using the ADL2 calls works for RX 400/500, but not for older cards, which still requires the old ADL calls. You'd have to kinda cascade every possible version of the ADL functions from newest to oldest until one registers something to support every possible generation.

The ADL2 calls(latest LHM build) worked for R9 280x ([here](https://github.com/Rem0o/FanControl.Releases/issues/2465#issuecomment-2111056143)), it also should work with R9 390. I'm not sure if the R9 390 in this case is using PMlog or if it is using overdrive. Maybe I need to change this [condition](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/3959e8e3c04452fed566e88fb40d0847a64581d8/LibreHardwareMonitorLib/Interop/AtiAdlxx.cs#L230). An LHM report would be helpful.

**Comment by weinitom** [comment]
> > @weinitom @epinter Might be, I believe collectively we tested with RX 400/500, RX 5000/6000/7000, but we didn't go older. The required gymnastic to support every generation might not be perfect yet. My guess @epinter is that using the ADL2 calls works for RX 400/500, but not for older cards, which still requires the old ADL calls. You'd have to kinda cascade every possible version of the ADL functions from newest to oldest until one registers something to support every possible generation.
> 
> The ADL2 calls(latest LHM build) worked for R9 280x ([here](https://github.com/Rem0o/FanControl.Releases/issues/2465#issuecomment-2111056143)), it also should work with R9 390. I'm not sure if the R9 390 in this case is using PMlog or if it is using overdrive. Maybe I need to change this [condition](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/3959e8e3c04452fed566e88fb40d0847a64581d8/LibreHardwareMonitorLib/Interop/AtiAdlxx.cs#L230). An LHM report would be helpful.

Can I help here?

**Comment by epinter** [comment]
> > > @weinitom @epinter Might be, I believe collectively we tested with RX 400/500, RX 5000/6000/7000, but we didn't go older. The required gymnastic to support every generation might not be perfect yet. My guess @epinter is that using the ADL2 calls works for RX 400/500, but not for older cards, which still requires the old ADL calls. You'd have to kinda cascade every possible version of the ADL functions from newest to oldest until one registers something to support every possible generation.
> > 
> > 
> > The ADL2 calls(latest LHM build) worked for R9 280x ([here](https://github.com/Rem0o/FanControl.Releases/issues/2465#issuecomment-2111056143)), it also should work with R9 390. I'm not sure if the R9 390 in this case is using PMlog or if it is using overdrive. Maybe I need to change this [condition](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/3959e8e3c04452fed566e88fb40d0847a64581d8/LibreHardwareMonitorLib/Interop/AtiAdlxx.cs#L230). An LHM report would be helpful.
> 
> Can I help here?

Download the latest nightly of LHM [here](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/actions/runs/9100036596/artifacts/1506168620), go to the file menu and save the report. Attach the text here. In Fan control remember to disable ADLX plugin, it's not supported on older cards.

**Comment by weinitom** [comment]
[LibreHardwareMonitor.Report.GPU.txt](https://github.com/Rem0o/FanControl.Releases/files/15366585/LibreHardwareMonitor.Report.GPU.txt)

**Comment by epinter** [comment]
> > > > > @weinitom @epinter Might be, I believe collectively we tested with RX 400/500, RX 5000/6000/7000, but we didn't go older. The required gymnastic to support every generation might not be perfect yet. My guess @epinter is that using the ADL2 calls works for RX 400/500, but not for older cards, which still requires the old ADL calls. You'd have to kinda cascade every possible version of the ADL functions from newest to oldest until one registers something to support every possible generation.
> > > > 
> > > > 
> > > > The ADL2 calls(latest LHM build) worked for R9 280x ([here](https://github.com/Rem0o/FanControl.Releases/issues/2465#issuecomment-2111056143)), it also should work with R9 390. I'm not sure if the R9 390 in this case is using PMlog or if it is using overdrive. Maybe I need to change this [condition](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/3959e8e3c04452fed566e88fb40d0847a64581d8/LibreHardwareMonitorLib/Interop/AtiAdlxx.cs#L230). An LHM report would be helpful.
> > > 
> > > 
> > > Can I help here?
> > 
> > 
> > Download the latest nightly of LHM [here](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/actions/runs/9100036596/artifacts/1506168620), go to the file menu and save the report. Attach the text here. In Fan control remember to disable ADLX plugin, it's not supported on older cards.
> 
> [LibreHardwareMonitor.Report.GPU.txt](https://github.com/Rem0o/FanControl.Releases/files/15366585/LibreHardwareMonitor.Report.GPU.txt)

Did you see the sensors in LHM ? They are correct in the report:

```
AMD GPU

Overdrive5 Parameters

Overdrive5 Temperature

 Status: ADL_OK
 Value: 59

Overdrive5 FanSpeed

 Status RPM: ADL_OK
 Value RPM: 4
 Status Percent: ADL_OK
 Value Percent: 0

OverdriveN Temperature

 Temperature[EDGE].Status: ADL_OK
 Temperature[EDGE].Value: 59
 Temperature[MEM].Status: ADL_OK
 Temperature[MEM].Value: 59
 Temperature[VRVDDC].Status: ADL_OK
 Temperature[VRVDDC].Value: 59
 Temperature[VRMVDD].Status: ADL_OK
 Temperature[VRMVDD].Value: 59
 Temperature[LIQUID].Status: ADL_OK
 Temperature[LIQUID].Value: 59
 Temperature[PLX].Status: ADL_OK
 Temperature[PLX].Value: 59
 Temperature[HOTSPOT].Status: ADL_OK
 Temperature[HOTSPOT].Value: 59
```

**Comment by weinitom** [comment]
![Screenshot 2024-05-18 213348](https://github.com/Rem0o/FanControl.Releases/assets/18164866/333e0ff5-df6f-49ff-b2bd-a890ac5c60d2)

Yes they are correct, the card is always around 60° in idle!

**Comment by weinitom** [comment]
Wait! After deleting config, redownloading and reboot I see the temperature now!

**Comment by qaziabdullah** [comment]
I am facing the same issue on rx570 4gb. Tried different builds but no use.

What happened was:
It was working fine with idk what version in past as I installed this a long time ago and don't remember what version was the application.
I took out my gpu and put in a rx 580 (2048 sp) to test that gpu for a friend
The temps were not working on that gpu but I ignored it as i was just testing the gpu

As soon as I installed my own rx 570 4gb back, the temps are not working on it.

The app opens with  "ADLX could not initialize or has no sensors."
LHM doesn't show the gpu temperature sensor as well.
I checked the nightly build @epinter mentioned and also the build that @Rem0o mentioned but the temperature sensor isn't showing.

Any idea what went wrong? 

btw Thanks for the amazing app. Really helps with controlling fans properly.
